### PR TITLE
[DOCS] Rewrite `regexp` query

### PR DESCRIPTION
**Pull request by jrodewig**
## Changes
* Rewrites description of `regexp` query
* Breaks **regular expression syntax** section into separate page
* Include all parameters in example request
* Adds parameters sections

This is part of #40977, an effort to standardize documentation for query types.

## Before
<details>
 <summary>Regexp query - Before image</summary>
Regexp query - Before image
</details>

## After
<details>
 <summary>Regexp query - After image</summary>
Regexp query - After image
</details>

<details>
 <summary>Regular expression syntax - After image</summary>
Regular expression syntax - After image
</details>

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/1

File: docs/reference/query-dsl/regexp-syntax.asciidoc [hunk 1]
```diff
@@ -1,286 +1,224 @@
...
+{es} uses https://lucene.apache.org/core/[Apache Lucene]'s regular expression
+engine to parse these queries.
 
-Any Unicode characters may be used in the pattern, but certain characters
-are reserved and must be escaped.  The standard reserved characters are:
+[float]
+[[regexp-reserved-characters]]
+=== Reserved characters
+Lucene's regular expression engine supports all Unicode characters. However, the
+following characters are reserved as operators:
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/regexp-syntax.asciidoc]: should we also add "and must be escaped"

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jrodewig

File: docs/reference/query-dsl/regexp-syntax.asciidoc [hunk 2]
```diff
@@ -1,286 +1,224 @@
...
-The question mark `"?"` makes the preceding shortest pattern optional. It
-matches zero or one times.  For string `"aaabbb"`:
-
-    aaa?bbb?    # match
-    aaaa?bbbb?  # match
-    .....?.?    # match
-    aa?bb?      # no match
-
+....
+abc     # matches 'ab', 'abb', 'abbb', 'abbbb', etc.
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/regexp-syntax.asciidoc]: Thanks for catching that @mayya-sharipova. Fixed with 19d387e.

**Pull request merged**